src/subject_name/verify.rs

```diff
@@ -317,45 +312,25 @@ impl<'a> Iterator for NameIterator<'a> {
     }
 }
 
-#[cfg(feature = "alloc")]
 pub(crate) fn list_cert_dns_names<'names>(
     cert: &'names crate::EndEntityCert<'names>,
-) -> Result<impl Iterator<Item = GeneralDnsNameRef<'names>>, Error> {
+) -> impl Iterator<Item = GeneralDnsNameRef<'names>> {
     let cert = &cert.inner();
-    let mut names = Vec::new();
-
-    let result =
-        NameIterator::new(Some(cert.subject), cert.subject_alt_name).find_map(&mut |result| {
-            let name = match result {
-                Ok(name) => name,
-                Err(err) => return Some(err),
-            };
-
-            let presented_id = match name {
-                GeneralName::DnsName(presented) => presented,
-                _ => return None,
-            };
-
-            let dns_name = DnsNameRef::try_from_ascii(presented_id.as_slice_less_safe())
-                .map(GeneralDnsNameRef::DnsName)
-                .or_else(|_| {
-                    WildcardDnsNameRef::try_from_ascii(presented_id.as_slice_less_safe())
-                        .map(GeneralDnsNameRef::Wildcard)
-                });
-
-            // if the name could be converted to a DNS name, add it; otherwise,
-            // keep going.
-            if let Ok(name) = dns_name {
-                names.push(name)
-            }
+    NameIterator::new(Some(cert.subject), cert.subject_alt_name).filter_map(|result| {
+        let name = match result {
+            Ok(name) => name,
+            // Ignore invalid names, this is an informational function, not something used
+            // during validation.
+            Err(_) => return None,
+        };
```

Comment:
nit,  take it or leave it: isn't this just
```suggestion
        // Ignore invalid names, this is an informational function, not something used
        // during validation.
        let name = result.ok()?;
```

Reply:
Fixed, thanks!